dash_table/Format.py

```diff
@@ -0,0 +1,200 @@
+from enum import Enum
```

Comment:
Def looking for input on the Python portions!
Here, mostly trying to hide away complexity through an interface that exposes the different aspects independently and through enums instead of the actual d3-format values.

The hope is to create an abstraction that would survive a switch from d3-format to some other library without having to rewrite the user facing contract.